src/asm_syntax.hpp

```diff
@@ -305,7 +305,13 @@ struct Assert {
 
 using Instruction = std::variant<Undefined, Bin, Un, LoadMapFd, Call, Exit, Jmp, Mem, Packet, LockAdd, Assume, Assert>;
 
-using LabeledInstruction = std::tuple<label_t, Instruction>;
+using btf_line_info = std::tuple<
+    std::string /* File Name */,
+    std::string /* Source Line */,
+    uint32_t    /* Line Number */,
+    uint32_t    /* Column Number */>;
+
+using LabeledInstruction = std::tuple<label_t, Instruction, std::optional<btf_line_info>>;
```

Comment:
@elazarg or @dthaler Is this the right spot to attach the line information? I wasn't sure if it should be attached to the LabeledInstruction or if I should add it to label_t? 

The reason I am asking is that this function:
https://github.com/vbpf/ebpf-verifier/blob/222214bef0eec4a56ff5b9fd8ed668cd3fb42e0d/src/asm_cfg.cpp#L39
converts LabeledInstruction  into a label_t and a Instruction, so the information is currently getting lost. If we move it to label_t and update label_t::operator<< then I think we will get the line info in the analysis report.

Reply:
If I move it to label_t, I can then modify print_report to do something like this:
```
alanjo@alanjo-dev:~/ebpf-verifier$ ./check -f ebpf-samples/build/packet_overflow.o

; /home/alanjo/ebpf-verifier/ebpf-samples/src/packet_overflow.c:24
;     int value = *(int*)data;
3:4: Upper bound must be at most packet_size (valid_access(r2.offset, width=4))
; /home/alanjo/ebpf-verifier/ebpf-samples/src/packet_overflow.c:24
;     int value = *(int*)data;
3:4: Upper bound must be at most packet_size (valid_access(r2.offset, width=4))
3:8: Code is unreachable after 3:8
```
Is this something that would be helpful?